lightkurve/correctors/designmatrix.py

```diff
@@ -379,44 +429,396 @@ def __getitem__(self, key):
+            dm = self.copy()
+
+        idx, jdx, v = find(dm.X)
+        weights = dm.X.copy()
+        weights[dm.X != 0] = 1
+        mean = np.bincount(jdx, weights=v)/np.bincount(jdx)
+        std = np.asarray([((np.sum((v[jdx == i] - mean[i])**2) * (1/((jdx == i).sum() - 1))))**0.5 for i in np.unique(jdx)])
+        mean[std == 0] = 0
+        std[std == 0] = 1
+        white = (dm.X - vstack([lil_matrix(mean)] * dm.shape[0])).multiply(vstack([lil_matrix(1/std)] * dm.shape[0]))
+        dm.X = white.multiply(weights)
+        return dm
+
+    def pca(self, nterms=6, **kwargs):
+        """Returns a new `.SparseDesignMatrix` with a smaller number of regressors.
+
+        This method will use Principal Components Analysis (PCA) to reduce
+        the number of columns in the matrix.
+
+        Parameters
+        ----------
+        nterms : int
+            Number of columns in the new matrix.
+
+        Returns
+        -------
+        `.SparseDesignMatrix`
+            A new design matrix with PCA applied.
+        """
+        # nterms cannot be langer than the number of columns in the matrix
+        if nterms > self.shape[1]:
+            nterms = self.shape[1]
+        # We use `fbpca.pca` instead of `np.linalg.svd` because it is faster.
+        # Note that fbpca is randomized, and has n_iter=2 as default,
+        # we find this to be too few, and that n_iter=10 is still fast but
+        # produces more stable results.
+        from fbpca import pca  # local import because not used elsewhere
+        new_values, _, _ = pca(self.values, nterms, n_iter=10)
+        return SparseDesignMatrix(type(self.X)(new_values), name=self.name, **kwargs)
+
+    def append_constant(self, prior_mu=0, prior_sigma=np.inf, inplace=False):
+        """Returns a new `.SparseDesignMatrix` with a column of ones appended.
+
+        Returns
+        -------
+        `.SparseDesignMatrix`
+            New design matrix with a column of ones appended. This column is
+            named "offset".
+        """
+        if inplace:
+            dm = self
+        else:
+            dm = self.copy()
+        dm.X = hstack([dm.X, lil_matrix(np.ones(dm.shape[0])).T], format='lil')
+        dm.prior_mu = np.append(dm.prior_mu, prior_mu)
+        dm.prior_sigma = np.append(dm.prior_sigma, prior_sigma)
+        return dm
+
+    def __getitem__(self, key):
+        loc = np.where(np.asarray(self.columns) == key)[0]
+        if len(loc) == 0:
+            raise ValueError('No such column as `{}`.'.format(key))
+        return self.X[:, loc].toarray()
+
+    @property
+    def shape(self):
+        """Tuple specifying the shape of the matrix as (n_rows, n_columns)."""
+        return self.X.shape
+
+    def __repr__(self):
+        return '{} SparseDesignMatrix {}'.format(self.name, self.shape)
+
+    def join(self, matrix):
+        """ Join two designmatrices, return a design matrix collection """
+        return SparseDesignMatrixCollection([self, matrix])
```

Comment:
`join` can be misunderstood for concatenation or cross product in this context.  Do we need this method?  Is it useful?

Reply:
The purpose of join was to reduce this line

`lk.SparseDesignMatrix([dm1, dm2])`
to
`dm1.join(dm2)`

Reply:
I wonder if calling `lk.SparseDesignMatrixCollection([dm1, dm2])` is actually better here, because it's not intuitive to me that `dm1.join(x)` returns a collection rather than just another design matrix?

Reply:
This has been changed to collect